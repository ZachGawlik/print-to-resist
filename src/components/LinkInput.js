import React, { PropTypes } from 'react';

export default class LinkInput extends React.Component {
  static propTypes = {
    index: PropTypes.number.isRequired,
    onUrlChange: PropTypes.func.isRequired,
    onDeleteLink: PropTypes.func.isRequired,
    url: PropTypes.string.isRequired
  };

  handleUrlChange = (event) => {
    this.props.onUrlChange(this.props.index, event.target.value);
  }

  handleDeleteClick = () => {
    this.props.onDeleteLink(this.props.index)
  }

  render() {
    return (
      <div>
        <input
          type="url"
          onChange={this.handleUrlChange}
          value={this.props.url}
        />
        <button type="button" onClick={this.handleDeleteClick}>
          -
        </button>
      </div>
    );
  }
}
