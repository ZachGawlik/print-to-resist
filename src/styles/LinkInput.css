.LinkInput {
  margin-bottom: 1em;
}

.LinkInput__remove {
  font-size: 16px;
  padding-top: 3px;
}
