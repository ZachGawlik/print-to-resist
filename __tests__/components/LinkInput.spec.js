import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LinkInput from '../../src/components/LinkInput';

describe('LinkInput', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <LinkInput
        index={1}
        onUrlChange={jest.fn()}
        onDeleteLink={jest.fn()}
        url="google.com"
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
