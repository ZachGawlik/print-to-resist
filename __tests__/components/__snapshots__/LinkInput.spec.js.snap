// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkInput renders as expected 1`] = `
<div
  className="LinkInput"
>
  <input
    onChange={[Function]}
    type="url"
    value="google.com"
  />
  <button
    className="LinkInput__remove"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
</div>
`;
